corehq/apps/users/templates/users/partials/edit_role_modal.html

```diff
@@ -25,6 +25,8 @@ <h4 class="modal-title" data-bind="text: modalTitle"></h4>
                        type="text"
                        id="role-name"
                        class="form-control"
+                       pattern="\S.*"
```

Comment:
How about also disallowing trailing whitespace?

Reply:
Good tip! The regex `^\S.*\S$` seems to work great for that. Ethan mentioned that it might be a better user experience to just remove the whitespace for them. What do you think?

Reply:
Cool. I like the thought of removing the whitespace automatically. It's generally not great to interfere with user input - it's confusing for users if they notice, and insanely frustrating if they didn't want that interference. However, leading/trailing whitespace is almost certainly an error.

I spent an hour just last week troubleshooting a problem with SMSes being broken for a project, which turned out to be due to a space at the end of an API key, innocently pasted in and nearly impossible to notice. So this issue is currently near to my heart 😛

Reply:
That SMS issue sounds very frustrating :( 
Sounds good though, I'll look into removing the whitespace automatically.

Reply:
> which turned out to be due to a space at the end of an API key, innocently pasted in and nearly impossible to notice

There's an in Django 1.x where it can't parse the geos version string (the format of which is patently ridiculous).  Whenever I update packages I have to edit the Django source code to allow a trailing space in the regex that pattern matches that version string.

Reply:
This? https://stackoverflow.com/questions/18643998/geodjango-geosexception-error

That sounds awful.

Reply:
That's the one.  Thankfully I've only had to fiddle with that twice so far, otherwise I'd probably sink more time into that than it's worth.